branching_xblock/static/js/src/studio_editor.js

```diff
@@ -1,172 +1,230 @@
-
-    function saveScenario() {
-        const nodes = [];
-        $element.find('.node').each(function() {
-            const $node = $(this);
-            const choices = [];
-            $node.find('.choice').each(function() {
-                choices.push({
-                    text: $(this).find('.choice-text').val(),
-                    target_node_id: $(this).find('.choice-target').val()
-                });
-            });
-            nodes.push({
-                id: $node.data('node-id'),
-                type: $node.find('.node-type').val(),
-                content: $node.find('.node-content').val(),
-                media: { url: $node.find('.media-url').val() },
-                choices: choices
-            });
-        });
-        runtime.notify('save', { state: 'saving' });
-        $.ajax({
-            type: 'POST',
-            url: runtime.handlerUrl(element, 'save_scenario'),
-            data: JSON.stringify({ nodes: nodes }),
-            success: (response) => {
-              if (response.success) {
-                runtime.notify('save', { state: 'saved' });
-              } else {
-                showErrors(response.errors);
-              }
-            }
-        });
-    }
-
-    function handleAddChoice(evt) {
-        const $choiceList = $(evt.target).closest('.choices-list');
-        $choiceList.append(`
-          <div class="choice">
-            <input type="text" class="choice-text" placeholder="Choice text">
-            <input type="text" class="choice-target" placeholder="Target Node ID">
-            <button class="delete-choice">x</button>
+
+    $editor.append(
+      '<button type="button" class="btn-add-node">Add Node</button>'
+    );
+
+    bindInteractions();
+    bindActions($settings);
+  }
+
+  function bindInteractions() {
+    $editor.find('.btn-delete-node').off('click').on('click', function() {
+      $(this).closest('.node-block').remove();
+      $editor.find('.node-block').each((i, el) => {
+          $(el).attr('data-node-idx', i)
+               .find('.node-title').text(`Node ${i+1}`);
+      });
+    });
+
+    $editor.find('.btn-add-choice').off('click').on('click', function() {
+      const $container = $(this).closest('.choices-container');
+      const currentNodeId = $(this).closest('.node-block').data('node-id');
+      const nodeOptions = $editor.find('.node-block').map((j, nb) => ({
+        id:  $(nb).data('node-id'),
+        label: `Node ${j+1}`
+      })).get().filter(opt => opt.id !== currentNodeId);
+      const renderOpts = sel => nodeOptions.map(opt =>
+        `<option value="${opt.id}" ${opt.id===sel?'selected':''}>${opt.label}</option>`
+      ).join('');
+      $container.append(`
+        <div class="choice-row">
+          <input class="choice-text" placeholder="Choice text"/>
+          <select class="choice-target">
+            ${ renderOpts('') }
+          </select>
+          <button type="button" class="btn-delete-choice">x</button>
+        </div>
```

Comment:
Instead of inline HTML like this, you should look into handlebars templates. 

The [poll XBlock](https://github.com/open-craft/xblock-poll) is a good example.

Reply:
Oh I wasn't aware of that. I will refactor the code to use handlebars templates. Thanks!

Reply:
Done.